src/Compilers/CSharp/Portable/Symbols/TypeParameterSymbol.cs

```diff
@@ -476,6 +500,24 @@ internal bool GetIsReferenceType(ConsList<TypeParameterSymbol> inProgress)
 
         public sealed override bool IsReferenceType => GetIsReferenceType(ConsList<TypeParameterSymbol>.Empty);
 
+        internal bool GetIsNotNullableIfReferenceType(ConsList<TypeParameterSymbol> inProgress)
+        {
+            if (inProgress.ContainsReference(this))
+            {
+                return false;
+            }
+
+            if (this.HasReferenceTypeConstraint && !this.HasNullableReferenceTypeConstraint)
+            {
+                return true;
+            }
+
+            return IsNotNullableIfReferenceTypeFromConstraintTypes(this.GetConstraintTypesNoUseSiteDiagnostics(inProgress, early: true), inProgress);
+        }
+
+        // PROTOTYPE(NullableReferenceTypes): Should this API be exposed through ITypeParameterSymbol?
```

Comment:
>Should this API be exposed through ITypeParameterSymbol? [](start = 46, length = 56)

Can consumers of the compiler API calculate this from the other properties including `HasNullableReferenceTypeConstraint`?

Reply:
>Can consumers of the compiler API calculate this from the other properties including HasNullableReferenceTypeConstraint?

>If so, we could skip exposing it, or provide an extension method.

I will leave the comment for now.


---
In reply to: [211450386](https://github.com/dotnet/roslyn/pull/29407#discussion_r211450386) [](ancestors = 211450386)